### ISSUE TITLE
Error when building from source on Linux

### ISSUE THREAD

**Issue by Captain-Barge** [SoundChaser83/devilutionx-apocrypha#12]
**Important information**
Compiling Apocrypha myself at commit 109d02faf2929c233b125435931f5bd712521a42

**Describe**
Error when compiling on AMD64 Ubuntu Linux 22.04.  Error messages at compile are below.

**To Reproduce**
Follow [build instructions](https://github.com/SoundChaser83/devilutionx-apocrypha/blob/SoundChaserBranch/docs/building.md):
```
# Install dependencies
sudo apt-get install cmake g++ libsdl2-dev libsodium-dev libpng-dev libbz2-dev libgtest-dev libgmock-dev libsdl2-image-dev libfmt-dev

# Get source code
git clone https://github.com/SoundChaser83/devilutionx-apocrypha.git
cd devilutionx-apocrypha/

# Build game
cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=Release
cmake --build build -j $(getconf _NPROCESSORS_ONLN)
```

**Expected behavior**
Game builds to `./devilutionx-apocrypha/build/` without errors

**Error written to terminal**
```
[ 63%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/drlg_l2.cpp.o
[ 63%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/drlg_l3.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/drlg_l4.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/dx.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/encrypt.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/engine.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/error.cpp.o
[ 64%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/gamemenu.cpp.o
[ 66%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/gendung.cpp.o
[ 66%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/gmenu.cpp.o
[ 66%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/help.cpp.o
[ 66%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/hwcursor.cpp.o
[ 66%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/init.cpp.o
[ 67%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/interfac.cpp.o
[ 67%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/inv.cpp.o
[ 67%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/itemdat.cpp.o
[ 67%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/items.cpp.o
[ 67%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/lighting.cpp.o
[ 68%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/loadsave.cpp.o
[ 68%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/menu.cpp.o
[ 68%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/minitext.cpp.o
[ 68%] Building CXX object CMakeFiles/libdevilutionx.dir/Source/misdat.cpp.o
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
  228 | };
      | ^
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
/home/landon/Downloads/devilutionx-apocrypha/Source/itemdat.cpp:228:1: error: narrowing conversion of ‘-1’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
gmake[2]: *** [CMakeFiles/libdevilutionx.dir/build.make:440: CMakeFiles/libdevilutionx.dir/Source/itemdat.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/home/landon/Downloads/devilutionx-apocrypha/Source/items.cpp:1593:6: warning: ‘void devilution::{anonymous}::Hex2bin(const char*, int, uint8_t*)’ defined but not used [-Wunused-function]
 1593 | void Hex2bin(const char *src, int bytes, uint8_t *target)
      |      ^~~~~~~
/home/landon/Downloads/devilutionx-apocrypha/Source/loadsave.cpp: In function ‘void devilution::LoadLevel()’:
/home/landon/Downloads/devilutionx-apocrypha/Source/loadsave.cpp:2230:80: warning: enumerated and non-enumerated type in conditional expression [-Wextra]
 2230 |                                 AutomapView[i][j] = automapView == MAP_EXP_OLD ? MAP_EXP_SELF : automapView;
      |                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[1]: *** [CMakeFiles/Makefile2:329: CMakeFiles/libdevilutionx.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

Edit: fixed typo

**Comment by SoundChaser83** [comment]
I'll have to show this to the DevilutionX Discord and see what they have to say about it. I appreciate your interest though!

**Comment by AJenbo** [comment]
Change -1 to 255

**Comment by Captain-Barge** [comment]
AJenbo's suggestion worked to get the code compiling and the game runs and works. Superficially this seems to resolve the issue. However, I'm not familiar with whether changing the `iMinMLvl` values from -1 to 255 has any gameplay consequences (maybe it goes against the spirit of the mod?).

I've opened a pull request with the tested change in case it's helpful. But maybe SoundChaser83 would rather not merge because there's a better way to go about this?

**Comment by SoundChaser83** [comment]
Closing this as the issue has been fixed via #13.
